scripts/deploy-buildkite-darwin/buildkite.hs

```diff
@@ -0,0 +1,214 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i runhaskell
+{-# LANGUAGE OverloadedStrings, LambdaCase #-}
+
+import Prelude hiding (FilePath)
+import Turtle
+import Data.Text (Text)
+import qualified Data.Text as T
+import qualified Data.Text.IO as T
+import System.IO (hFlush)
+import System.Environment (getExecutablePath)
+import Filesystem.Path.CurrentOS (decodeString)
+import Control.Monad (forM_)
+
+-- https://github.com/LnL7/nix-darwin/pull/70
+darwinSource :: Text
+darwinSource = "https://github.com/LnL7/nix-darwin/archive/5869c4e45bcfd402ba4f5db1489543ab79358f94.tar.gz"
+
+nixpkgsSource :: Text
+nixpkgsSource = nixChannel <> "/nixexprs.tar.xz"
+
+nixChannel :: Text
+nixChannel = "https://nixos.org/channels/nixpkgs-17.09-darwin"
+
+iohkOpsSource :: Text
+iohkOpsSource = "https://github.com/input-output-hk/iohk-ops/archive/master.tar.gz"
+
+main :: IO ()
+main = sh $ do
+  -- install nix-darwin
+  checkNix
+  setupSSLCert
+  setupNixChannel nixChannel
+  nixPath <- setupNixPath
+  installNixDarwin darwinSource
+
+  -- set up private buildkite keys dir then switch to new configuration
+  bkDir <- (</> "buildkite") <$> home
+  dataPath <- liftIO getDataPath
+  setupConfiguration (dataPath </> "darwin-configuration.nix") bkDir nixPath
+  setupBuildkiteDir bkDir
+  darwinRebuild
+  secureBuildkiteDir bkDir
+
+  -- enable xcode command line developer tools because some things in
+  -- the npm build require it.
+  enableXCodeCL
```

Comment:
Huh, `npm build` of daedalus? I don't think we have xcode installed on any of the existing macos so I'm surprised about this.